src/auth/AxiosJwtAuthService.js

```diff
@@ -0,0 +1,303 @@
+   * logged in.
+   */
+  async fetchAuthenticatedUser() {
+    const decodedAccessToken = await this.jwtTokenService.getJwtToken();
+
+    if (decodedAccessToken !== null) {
+      this.setAuthenticatedUser({
+        userId: decodedAccessToken.user_id,
+        username: decodedAccessToken.preferred_username,
+        roles: decodedAccessToken.roles || [],
+        administrator: decodedAccessToken.administrator,
+      });
+    }
+
+    return this.getAuthenticatedUser();
+  }
+
+  /**
+   * Ensures a user is authenticated. It will redirect to login when not
+   * authenticated.
+   *
+   * @param {string} [redirectUrl=config.BASE_URL] to return user after login when not
+   * authenticated.
+   * @returns {Promise<UserData>}
+   */
+  async ensureAuthenticatedUser(redirectUrl = this.config.BASE_URL) {
+    await this.fetchAuthenticatedUser();
+
+    if (this.getAuthenticatedUser() === null) {
+      const isRedirectFromLoginPage = global.document.referrer &&
+        global.document.referrer.startsWith(this.config.LOGIN_URL);
+
+      if (isRedirectFromLoginPage) {
+        const redirectLoopError = new Error('Redirect from login page. Rejecting to avoid infinite redirect loop.');
+        logFrontendAuthError(this.loggingService, redirectLoopError);
+        throw redirectLoopError;
+      }
+
+      // The user is not authenticated, send them to the login page.
+      this.redirectToLogin(redirectUrl);
+
+      const unauthorizedError = new Error('Failed to ensure the user is authenticated');
+      unauthorizedError.isRedirecting = true;
+      throw unauthorizedError;
+    }
+
+    return this.getAuthenticatedUser();
+  }
+
+  /**
+   * Fetches additional user account information for the authenticated user and merges it into the
+   * existing authenticatedUser object, available via getAuthenticatedUser().
+   *
+   * ```
+   *  console.log(authenticatedUser); // Will be sparse and only contain basic information.
+   *  await hydrateAuthenticatedUser()
+   *  const authenticatedUser = getAuthenticatedUser();
+   *  console.log(authenticatedUser); // Will contain additional user information
+   * ```
+   *
+   * @returns {Promise<null>}
+   */
+  async hydrateAuthenticatedUser() {
+    const user = this.getAuthenticatedUser();
+    if (user !== null) {
+      const response = await this.authenticatedHttpClient
+        .get(`${this.config.LMS_BASE_URL}/api/user/v1/accounts/${user.username}`);
+      this.setAuthenticatedUser({ ...user, ...camelCaseObject(response.data) });
+    }
+  }
+
+  /**
+ * Adds authentication defaults and interceptors to an HTTP client instance.
+ *
+ * @param {HttpClient} newHttpClient
+ * @param {Object} config
+ * @param {string} [config.REFRESH_ACCESS_TOKEN_ENDPOINT]
+ * @param {string} [config.ACCESS_TOKEN_COOKIE_NAME]
+ * @param {string} [config.CSRF_TOKEN_API_PATH]
+ * @returns {HttpClient} A configured Axios HTTP client.
+ */
+  addAuthenticationToHttpClient(newHttpClient) {
```

Comment:
This function takes the HTTP client to apply authentication to, but otherwise uses the configuration of this AxiosJwtAuthService instance to do it.  That feels like a reasonable method signature to me - it used to be a static function, but that was back when everything was static/singletons.